source/common/common/byte_order.h

```diff
@@ -25,3 +25,127 @@
 #include <endian.h>
 
 #endif
+
+enum class ByteOrder { Host, LittleEndian, BigEndian };
```

Comment:
I see you don't support middle endian architectures ;) The PDP-11 port will have to wait.